File: septunneuf-serveur/septunneuf-monsters/src/main/java/fr/ebiron/septunneuf/monsters/config/RabbitMQConfig.java
package fr.ebiron.septunneuf.monsters.config;



import org.springframework.amqp.core.Queue;
import org.springframework.amqp.core.QueueBuilder;
import org.springframework.amqp.rabbit.connection.ConnectionFactory;
import org.springframework.amqp.rabbit.core.RabbitTemplate;
import org.springframework.amqp.support.converter.Jackson2JsonMessageConverter;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.HashMap;
import java.util.Map;

@Configuration
public class RabbitMQConfig {

    @Bean
    public Queue addToInventoryQueue(){
        return QueueBuilder.durable("addMonsterToInventory.queue")
                .deadLetterExchange("addMonsterToInventory.exchange")
                .deadLetterRoutingKey("addMonsterToInventory.queue.wait")
                .build();
    }

    @Bean
    public Queue createMonsterQueue(){
        return new Queue("createMonster.queue");
    }

    @Bean
    public Queue deleteMonsterQueue(){
        return new Queue("deleteMonster.queue");
    }

    @Bean
    public RabbitTemplate rabbitTemplate(final ConnectionFactory connectionFactory) {
        final RabbitTemplate rabbitTemplate = new RabbitTemplate(connectionFactory);
        rabbitTemplate.setMessageConverter(producerJackson2MessageConverter());
        return rabbitTemplate;
    }

    @Bean
    public Jackson2JsonMessageConverter producerJackson2MessageConverter() {
        return new Jackson2JsonMessageConverter();
    }
}

File: septunneuf-serveur/septunneuf-eggs/src/main/kotlin/fr/ebiron/septunneuf/eggs/config/RabbitMQConfig.kt
package fr.ebiron.septunneuf.eggs.config

import org.springframework.amqp.core.Queue
import org.springframework.amqp.rabbit.connection.ConnectionFactory
import org.springframework.amqp.rabbit.core.RabbitTemplate
import org.springframework.amqp.support.converter.Jackson2JsonMessageConverter
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration


@Configuration
class RabbitMQConfig {
    @Bean
    fun removeEggsQueue(): Queue {
        return Queue("removeEggs.queue")
    }

    @Bean
    fun rabbitTemplate(connectionFactory: ConnectionFactory): RabbitTemplate {
        val rabbitTemplate = RabbitTemplate(connectionFactory)
        rabbitTemplate.messageConverter = producerJackson2MessageConverter()
        return rabbitTemplate
    }

    @Bean
    fun producerJackson2MessageConverter(): Jackson2JsonMessageConverter {
        return Jackson2JsonMessageConverter()
    }
}